markets/perps-market/contracts/modules/LiquidationModule.sol

```diff
@@ -160,27 +163,44 @@ contract LiquidationModule is ILiquidationModule, IMarketEvents {
             runtime.loopIterator < openPositionMarketIds.length;
             runtime.loopIterator++
         ) {
-            uint128 positionMarketId = openPositionMarketIds[runtime.loopIterator].to128();
-            uint256 price = PerpsPrice.getCurrentPrice(
-                positionMarketId,
+            runtime.positionMarketId = openPositionMarketIds[runtime.loopIterator].to128();
+            runtime.price = PerpsPrice.getCurrentPrice(
+                runtime.positionMarketId,
                 PerpsPrice.Tolerance.STRICT
             );
 
             (
                 uint256 amountLiquidated,
                 int128 newPositionSize,
                 int128 sizeDelta,
+                uint256 positionFlaggedAmount,
                 MarketUpdate.Data memory marketUpdateData
-            ) = account.liquidatePosition(positionMarketId, price);
+            ) = account.liquidatePosition(runtime.positionMarketId, runtime.price);
+
+            // using amountToLiquidate to calculate liquidation reward
+            runtime.liquidationReward = includeFlaggingReward
+                ? PerpsMarketConfiguration
+                    .load(runtime.positionMarketId)
+                    .calculateLiquidationReward(positionFlaggedAmount.mulDecimal(runtime.price))
+                : 0;
```

Comment:
it might be easier to understand what's happening her if just calculated liquidation rewards here regardless of `includeFlagginReward` and check the flag after.

Reply:
did a change on the logic (just renaming variables and moving small pieces of code to a different place). Please reassess.